core/src/main/java/hudson/model/queue/Executables.java

```diff
@@ -59,6 +71,49 @@
             }
         }
     }
+    
+    /**
+     * Get parent subtask from which the executable has been created.
+     * @param e Executable.
+     * @return Parent subtask from which the executable has been created.
+     *         {@code null} if the Executable is {@code null} OR has incompatible API (old plugin depending on a core older than 1.377)
+     * @since TODO
+     * @see #getParentOfOrFail(hudson.model.Queue.Executable) 
+     */
+    @CheckForNull
+    public static SubTask getParentOfOrNull(@CheckForNull Executable e) {
+        if (e == null) {
+            return null;
```

Comment:
Please do not introduce API methods accepting null arguments for no particular reason. Should be `@Nonnull`.